### PR TITLE
TOV in cowling approximation

### PR DESCRIPTION
**Pull request by Yurlungur**
The current state of the TOV star test under the Cowling approximation. Things are currently sort of working.

### CODE REVIEW

**Comment by Yurlungur** [comment]
@jdolence @brryan @lroberts36 Here's where I'm at:

- I moved the TOV star initial conditions to a stable branch
- Immediately I found evolution to be stable, and right-hand-sides to be small. Now double-checked.
- This implies to me that the problem was that the formulation I chose doesn't support singularity/horizon formation, but otherwise works as intended.

I spoke to @jdolence and we came up with the following plan:
1. Review the PR as it stands and get it into main, so I can start making changes from a working, main branch
2. Work on getting Cylindrical and Cartesian coordinates more fully working
3. Reformulate the equations a bit to try and use conserved, rather than primitive variables as much as possible
4. Set up a dust collapse test.

Next step then is a review of what I currently have. Since I'll be making changes, please don't spend too much time on this. But I do want at least one other pair of eyes on this before I merge.

File: src/monopole_gr/monopole_gr.cpp [hunk 1]
```diff
@@ -174,6 +221,21 @@ TaskStatus IntegrateHypersurface(StateDescriptor *pkg) {
...
+	  Real r = radius.x(i);
+	  fprintf(pf,
+		  "%.14e %.14e %.14e\n",
+		  r,
+		  matter_h(0,i),
+		  matter_h(1,i));
+  }
+  fclose(pf);
+  */
+
```
Comment (jdolence) [inline src/monopole_gr/monopole_gr.cpp:238]: remove?

**Review by jdolence — APPROVED** [review]
I did _not_ give this a thorough review.  There's a lot here.

File: src/main.cpp [hunk 2]
```diff
@@ -89,6 +91,11 @@ int main(int argc, char *argv[]) {
   // This line actually runs the simulation
   auto driver_status = driver.Execute();
 
+  // output some diagnostics
+  if (parthenon::Globals::my_rank == 0) {
+    MonopoleGR::DumpCurrentState(pman.pmesh->packages.Get("monopole_gr").get());
+  }
+
```
[the remark this answers is not in the document]
Comment (Yurlungur) [reply src/main.cpp:98]: good catch this should be cleaned up

File: src/geometry/monopole.hpp [hunk 3]
```diff
@@ -42,6 +42,26 @@ namespace Geometry {
...
+void S2C(Real r, Real th, Real ph, Real J[NDFULL][NDFULL]);
+KOKKOS_INLINE_FUNCTION
+void C2S(Real x, Real y, Real z, Real r, Real J[NDFULL][NDFULL]);
+KOKKOS_INLINE_FUNCTION
+void Hessian(Real x, Real y, Real z, Real r, Real H[NDFULL][NDFULL][NDFULL]);
+} // namespace MonopoleCoordTransforms
+
 class MonopoleSph {
  public:
   friend class MonopoleCart;
```
[the remark this answers is not in the document]
Comment (Yurlungur) [reply src/geometry/monopole.hpp:67]: I will deal with this when I clean up the 3D stuff. That's only needed for the 3D.

**Comment by Yurlungur** [comment]
Thanks for the review @jdolence if you're happy should I go ahead and merge this in? I know there's a lot here... Happy to walk you through it before merge too.

**Pull request merged**